src/Service/ConfigurationFactory.php

```diff
@@ -4,6 +4,7 @@
 
 namespace DoctrineMongoODMModule\Service;
 
+use Doctrine\Common\Cache\Psr6\CacheAdapter;
```

Comment:
DoctrineModule currently allows either cache v1 or v2 [(see here)](https://github.com/doctrine/DoctrineModule/blob/6.1.x/composer.json#L43), so this change would bring a new, but probably acceptable limitation.

Anyways, when referring to this symbol, you need to add the respective require statement in composer.json.

Reply:
> Anyways, when referring to this symbol, you need to add the respective require statement in composer.json.

Ah right, no [composer-require-checker](https://github.com/maglnet/ComposerRequireChecker) here 😉

Reply:
> Then we will have a Laminas cache, wrapped as a doctrine cache, wrapped as a PSR6 cache. Not sure if that is desired. Preferably, Laminas cache would be PSR6 by default, but not sure what the current state there is.

Afaik there's no lamians-cache here, just doctrine-cache. The deprecated `setMetadataCacheImpl` wants a doctrine-cache implementation.

> I'm doing it manually 😝
> 
> Though I would appreciate adding it in all three modules. Last time I tried this, there were some issues with optional dependencies, where I would have to whithelist every symbol from an optional dependency - not really what I wanted to do, so I didn't look into this further.

OT, but you can work around that with adding those as dev-dependencies. You'll (hopefully) need the dev-depencency anyways because you need the package in tests.

Reply:
> Afaik there's no lamians-cache here, just doctrine-cache. The deprecated setMetadataCacheImpl wants a doctrine-cache implementation.

With DoctrineModule v6, there is. The caches you pull via `doctrine.cache.[name]` are wrapped Laminas instances.

Reply:
> I just wanted to add the dependency, but actually there's no direct usage of any PSR-6 symbol.

I did not meat PSR, but doctrine/cache. The interface you refer to is AFAIK only available in v2, not in v1, but we allow both versions (for easier transition).

Reply:
> > Afaik there's no lamians-cache here, just doctrine-cache. The deprecated setMetadataCacheImpl wants a doctrine-cache implementation.
> 
> With DoctrineModule v6, there is. The caches you pull via `doctrine.cache.[name]` are wrapped Laminas instances.
> 
> You can pull the unwrapped instances via `doctrinemodule.cache.[name]` though.

Ah, okay, I did not realize that. We have some customized factories here and the main cache (phpfile) has it's own factory, so I didn't get a wrapped laminas cache implementation
So theoretically, I could instead pull `doctrinemodule.cache.[name]` and wrap the laminas-cache in a psr-6 cache. But that would involve more changes (new `\DoctrineMongoODMModule\Options\Configuration::getLaminasMetadataCache()` and some wiring).
It's probably cleaner to get rid of all doctrine-cache stuff in a new major version and just use psr-6 everywhere?

Reply:
> > I just wanted to add the dependency, but actually there's no direct usage of any PSR-6 symbol.
> 
> I did not meat PSR, but doctrine/cache. The interface you refer to is AFAIK only available in v2, not in v1, but we allow both versions (for easier transition).

Ah you're right. `doctrine/cache` will be a direct dependency. But the PSR wrapper was introduced in [`1.11.0`](https://github.com/doctrine/cache/releases/tag/1.11.0) as a forward compatibility layer so we can require `"doctrine/cache": "^1.11 || ^2.0"`